Info_msg info
string[] type_cnstr
float64[] data 

